src/Common/src/Interop/Unix/System.Native/Interop.UName.cs

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Runtime.InteropServices;
+using System.Text;
+
+internal static partial class Interop
+{
+    internal static partial class Sys
+    {
+        /// <summary>
+        /// Get the name of the current system. 
+        /// </summary>
+        /// <param name="machine">The OS architecture value.</param>
+        /// <returns>
+        /// Returns non-negative value on success, -1 on failure.
+        /// </returns>
+        [DllImport(Libraries.SystemNative, CharSet = CharSet.Ansi, SetLastError = true)]
+        internal static extern int UName([Out] StringBuilder machine, int capacity);
+
+        internal static int UName(out string machine)
+        {
+            int maxBufferLength = 1024;
```

Comment:
How do we know that the data is going to be less than this in size?  If it's not the native implementation in this PR will truncate it and we could theoretically lose useful information.  Is this a documented upper bound?